HarmonyTests/Patching/Specials.cs

```diff
@@ -50,6 +50,34 @@ public void Test_HttpWebRequestGetResponse()
 			Assert.True(HttpWebRequestPatches.postfixCalled, "Postfix not called");
 		}
 
+		[Test]
+		public void Test_PatchResultRef()
+		{
+			var test = new ResultRefStruct();
+
+			var instance = new Harmony("result-ref-test");
+			Assert.NotNull(instance);
+			var processor = instance.CreateClassProcessor(typeof(ResultRefStruct_Patch));
+			Assert.NotNull(processor, "processor");
+
+			test.Method1() = 1;
+			test.Method2() = 2;
+			test.Method3() = 3;
+			Assert.Throws<Exception>(() => test.Method4(), "Method4 does not throw");
+
+			Assert.AreEqual(new[] { 1, 0, 2, 0, 3, 0, 0 }, ResultRefStruct.Ints);
```

Comment:
Running test multiple times might fail it because the zeros are not zero anymore. Better add an initial reset of `ResultRefStruct.Ints`